### ISSUE TITLE
$.offset() is returning improper top, left for scaled viewports on devices such as Chromebook Touch/Surface/iOS/Android for relatively positioned parents.

### ISSUE THREAD

**Issue by aroussetpsn** [jquery/jquery#3187]
### Description ###
I expect when returning the offset, even if the viewport is scaled, it should properly give me the top & left registration point of the relatively positioned parent element. Instead, the amount of offset in the window's pageYOffset/pageXOffset properties need to be subtracted out to give a proper top & left registration point for the relatively positioned parent. You also need to subtract out the document element's top & left values in its client rect in case it has been scrolled when panning around the scaled viewport.

This is how we're handling it in our code once we get the offset back from jQuery: 

```javascript
var offset = $(el).offset();

//get document element width without scrollbar
var prevStyle = document.body.style.overflow || "";
document.body.style.overflow = "hidden";
var docWidth = document.documentElement.clientWidth;
document.body.style.overflow = prevStyle;

if(docWidth / window.innerWidth !== 1) { //kind of a hack to determine if the viewport has been scaled
    var docRect = document.documentElement.getBoundingClientRect();
    offset = {
        top: offset.top - window.pageYOffset - docRect.top,
        left: offset.left - window.pageXOffset - docRect.left
    };
}
```
Note: We've only tested this in WebKit based browsers such as Mobile Safari & Chrome.

[JSBin](http://jsbin.com/jacuhakafa/)
__Testcase Steps__
* Open link on a touch device (Chromebook Touch/iOS/Android/Surface) in a WebKit browser (Safari/Chrome for example).
* Spread two fingers gesture to zoom page in.
* Tap the green box to place a red square within it.
* Notice that the red square is not placed within its parent correctly.

**Comment by dmethvin** [comment]
I think this problem is covered in [the API docs](http://api.jquery.com/offset/):

> dimensions may be incorrect when the page is zoomed by the user; browsers do not expose an API to detect this condition.

If the "hack" you mention is consistent enough across the supported jQuery 3.0 browsers we could try to tackle fixing this, it's been around since the beginning of time.

**Comment by aroussetpsn** [comment]
@dmethvin Thanks for the reply, I missed that when reading the docs. So, that's what I suspected. I called it a hack mostly, because while the correction of the offset I feel is pretty sound for the browsers we support, determining if there's a scaled viewport or not isn't really that strong. I'd much rather have some sort of event or API to check against. 

Anyway, our product supports a lot later browsers than the version of jQuery (1.x) we're using so we can get away with it. YMMV for other browsers that jQuery 3.0 supports.
